### DISCUSSION TITLE
ECH

### DISCUSSION THREAD

**Discussion by WeAreLiving1984** [hufrea/byedpi#225] [Q&A]
Что делать с блокировкой по ECH, почти каждый сайт использует cloudflare, даже обновить системные пакеты не могу. byedpi, как минимум, с моими параметрами на помогает. И что теперь смысл allow/white list'ов пропадает, если треть доменов в него потенциально попадают?

**Comment by hufrea** [comment]
> Что делать с блокировкой по ECH

ТСПУ блокирует ECH если в SNI `cloudflare-ech.com`. Так что задача та же, что и была - не дать спарсить SNI. К тому же, ECH можно и просто отключить.

>  И что теперь смысл allow/white list'ов пропадает

Можно использовать домен, указанный выше. Раз все сайты за Cloudflare, то для них можно использовать одну и ту же стратегию, ведь реагировать на разные методы они будут одинаково.

**Reply by microcoder** [reply]
А как он блокирует, если SNI в ECH зашифрован? Что-то не понятно

**Reply by mikeS7** [reply]
> А как он блокирует, если SNI в ECH зашифрован? Что-то не понятно

base64 это не шифрование, а простое кодирование )

**Reply by hufrea** [reply]
> А как он блокирует, если SNI в ECH зашифрован? Что-то не понятно

ClientHello содержит открытую и закрытую часть (та которая в ECH). В открытой хранится outer SNI (общий для всех сайтов за Cloudflare), в закрытой же хранится inner SNI (настоящий, на который идет обращение).

**Reply by microcoder** [reply]
Ах вот оно что! Понятно! Изучаю этот вопрос и ещё не дошел до таких тонкостей. Спасибо за разъяснение!
